modules/nodes-cluster-resource-configure-jdk.adoc

```diff
@@ -29,20 +29,21 @@ this documentation, and may involve setting multiple additional JVM options.
 == Understanding how to override the JVM maximum heap size
 
 For many Java workloads, the JVM heap is the largest single consumer of memory.
-Currently, the OpenJDK defaults to allowing up to 1/4 (1/`-XX:MaxRAMFraction`)
-of the compute node's memory to be used for the heap, regardless of whether the
-OpenJDK is running in a container or not. It is therefore *essential* to
+
+Currently, on bare metal or VM setups, the OpenJDK's default heap value is 25%, 1/4th of the value (1/`-XX:MaxRamPercentage`). However, within containers using cgroups detection (v1 and v2 for JDK 11/17), the Xmx setting uses 50% of the maximum memory limits. In ubi9 - RHEL 9 images, the default allocation is 80%, leaving 20% for native use. It is therefore *essential* to
 override this behavior, especially if a container memory limit is also set.
 
+The `UseContainerSupport` flag is on by default in the JVM. It helps the JVM recognize Docker container operations, automatically fetching container-specific details like CPU count and total memory. Therefore, it is not necessary to manually define `JAVA_MAX_INITIAL_MEM`, as the JVM obtains maximum memory allocation from the pods directly.
+
 There are at least two ways the above can be achieved:
```

Comment:
from https://access.redhat.com/solutions/4886081, MaxRAMFraction is deprecated(https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8186248), so I think here the first way make no sense. And in the second way, you should add "MaxRAMPercentage" (override one of -XX:MaxRAM, -XX:MaxHeapSize , -Xmx or MaxRAMPercentage)

Reply:
Let me remove the first method.

Reply:
@lyman9966 For the second method, is there any link or something that can be added or any commands?

Reply:
You can give an example similar to https://access.redhat.com/solutions/4886081 :
*Note: Prior to JDK17, MaxRAMPercentage is based on MaxRAM, not system memory. The default max value for MaxRAM is 128g. For example, this results in a 64g max heap, not 100g, in a container with a 200g memory limit: -XX:MaxRAMPercentage=50. A workaround for using MaxRAMPercentage in environments with >128g available memory is to explicitly set MaxRAM to a higher value. For example, this would result in a 100g max heap in a container with a 200g memory limit: -XX:MaxRAMPercentage=50 -XX:MaxRAM=200g). See [JDK-8222252](https://access.redhat.com/bounce/?externalURL=https%3A%2F%2Fbugs.openjdk.org%2Fbrowse%2FJDK-8222252).

Reply:
@lyman9966 How about adding 'IMPORTANT 'admonition here, to explain that note, something like this:-
"Prior to JDK17, note that `MaxRAMPercentage` used `MaxRAM`, instead of system memory. The default maximum value for `MaxRAM` is set to 128g. This means that in a container with a 200g memory limit, using `-XX:MaxRAMPercentage=50` will result in a 64g maximum heap, not 100g.

To address this issue in environments with more than 128g of available memory, a workaround is to explicitly set `MaxRAM` to a higher value. For instance, using `-XX:MaxRAMPercentage=50` `-XX:MaxRAM=200g` would result in a 100g maximum heap within a container with a 200g memory limit."

Reply:
I'm OK for this change!

Reply:
Great. Let me move forward with further process.